Intro: Add `jax.vmap` section

Smit-create opened the pull request:
https://github.com/QuantEcon/lecture-jax/pull/87#issuecomment-1667002764

jstac commented on lectures/jax_intro.md:
```diff
@@ -424,8 +425,77 @@ ax.legend(loc='upper center')
+Consider the following function
+
+$$
+     f(x, y)= \frac{cos(x^2 + y^2)}{(1 + x^2 + y^2)}
```
`cos` should be `\cos`

jstac commented on lectures/jax_intro.md:
```diff
@@ -424,8 +425,77 @@ ax.legend(loc='upper center')
+     f(x, y)= \frac{cos(x^2 + y^2)}{(1 + x^2 + y^2)}
+$$
+
+Assume that $x$ and $y$ are 2-D.
```
Please cut this line. ("Assume...")

Smit-create commented:
> Can you think of one?
> 
> Also, when you first explained this double vmap idea to me, you used a really simple example in one of your recent PRs --- I think the one about inventories. (You compared two different types of vmap and how they worked relative to a loop.) Could you please add that in too, at the start of the discussion.

I updated it with a very simple example, and now it's easy for newcomers to understand this.

> I think it would be better if you got rid of f_loops and showed how it would be possible to create a vectorized version of f that accepts flat arrays x and y.

I removed that example completely as broadcasting is very easy to understand, but we can compare loops and vectorized implementation when `f` isn't broadcastable as updated in the last commit.

jstac commented:
Beautiful, thanks @Smit-create !
